dataRetriever/requestHandlers/requestHandler.go

```diff
@@ -346,10 +354,22 @@ func (rrh *resolverRequestHandler) RequestTrieNodes(destShardID uint32, hashes [
 	if len(unrequestedHashes) == 0 {
 		return
 	}
+
+	rrh.mutexTrieHashes.Lock()
+	defer rrh.mutexTrieHashes.Unlock()
+
+	rrh.trieHashAccumulator = append(rrh.trieHashAccumulator, unrequestedHashes...)
+	elapsedTime := time.Since(rrh.lastTrieRequestTime)
+	if len(rrh.trieHashAccumulator) < minHashesToRequest && elapsedTime < timeToAccumulateTrieHashes {
```

Comment:
What if the last bulk of trieHashes is smaller than 10? Let say 8. This will make the method to return in this point. It would be called again for the same 8 hashes after a while? 
1) If yes, they would be added  again in rrh.trieHashAccumulator, and now we have 16, but they are duplicates. 
2) If not, they would be not requested anymore.

Reply:
I think you should either call: rrh.addRequestedItems(rrh.trieHashAccumulator) before return or better rrh.trieHashAccumulator = rrh.trieHashAccumulator[:0] and add them again in the next call

Reply:
There are also other exit points below so I think you should put this line: rrh.trieHashAccumulator = rrh.trieHashAccumulator[:0], in a defer method above and remove it from line 400

Reply:
the request trie hashes method is called in a continuous way at every second. Thus the accumulated hashes will be surely requested.

Reply:
Then you need only to manage the duplicate hashes added in rrh.trieHashAccumulator after each call

Reply:
done with MAP.